src/platform/silabs/wifi/wf200/WifiInterface.cpp

```diff
@@ -293,6 +293,17 @@ sl_status_t get_all_counters(void)
     return result;
 }
 
+/**
+ * @brief Convert RCPI to RSSI
+ * @param[in]  rcpi: Received Channel Power Indicator value
+ * @return RSSI value
+ */
+int16_t ConvertRcpiToRssi(uint32_t rcpi) {
```

Comment:
If we want to return errors, we shouldn't return an arbitrary value. The signature should be :
```
CHIP_ERROR ConvertRCPIToRssi(uint32_t rcpi, int16_t & rssi);
```

Then again, since this is a local function, it is good enough to just set the closest value to reduce the complexity.
In other words, if the result is < int16_min, set int16_min. If the result is >int16_max, set int16_max.

Reply:
done